packages/devtools-local-toolbox/public/js/index.js

```diff
@@ -0,0 +1,102 @@
+/* global window, document, DebuggerConfig */
+
+const { bindActionCreators, combineReducers } = require("redux");
+const { Provider } = require("react-redux");
+const ReactDOM = require("react-dom");
+const React = require("react");
+
+const DevToolsUtils = require("devtools-sham/shared/DevToolsUtils");
+const AppConstants = require("devtools-sham/sham/appconstants").AppConstants;
+const { injectGlobals, debugGlobal } = require("./utils/debug");
+const { isEnabled, getValue,
+        isDevelopment, setConfig } = require("../../config/feature");
+
+setConfig(DebuggerConfig);
+
+// Set various flags before requiring app code.
+if (isEnabled("logging.client")) {
+  DevToolsUtils.dumpn.wantLogging = true;
+}
+
+const client = require("./clients");
+const { getClient, connectClients, startDebugging } = client;
+
+const Root = require("./components/Root");
+
+function initApp() {
+  const configureStore = require("./utils/create-store");
+  const reducers = require("./reducers");
+  const LandingPage = require("./components/LandingPage");
+
+  const createStore = configureStore({
+    log: getValue("logging.actions"),
+    makeThunkArgs: (args, state) => {
+      return Object.assign({}, args, { client: getClient(state) });
+    }
+  });
+
+  const store = createStore(combineReducers(reducers));
+  const actions = bindActionCreators(require("./actions"), store.dispatch);
+
+  if (isDevelopment()) {
+    AppConstants.DEBUG_JS_MODULES = true;
+    injectGlobals({ store });
+  }
+
+  return { store, actions, LandingPage };
+}
+
+function renderRoot(_React, _ReactDOM, component, _store) {
```

Comment:
I don't think we should be depending on the toolbox to do all of this. I'd like to see it more as a library than a framework. Just give me the pieces to connect, and I'll hook them up to my app. It's not uncommon that we want to tweak things in here and we can't do that if this is shared across everyone.

I think it's very enlightening to just open up an app's `main.js` and see a very straight-forward way of how things are hooked up. If we bury this, it gets harder for beginners to understand what's going on (and even advanced users who want to hack up something quickly to change it).

For example, I frequently go into `main.js` and flip `log: true` in the store configuration to log actions. Why don't I change the config? It's just easier for me to open `main.js` (config is at the root) and more importantly I don't have to restart the server.

I think this could export reducers/actions/components that someone can embed in their app. Or at least make the toolbox have its own render and redux store, and do everything it needs to if we don't have a target. But in our `main.js`, if we have a target, call into the toolbox to connect it, but let us do our own rendering/store creation.

Do whatever you feel like for now, this is a big change so we can move things back. I'm not going to block it.